### DISCUSSION TITLE
Portafolio weight constraint problem

### DISCUSSION THREAD

**Discussion by LuEdu87** [cvxpy/cvxpy#1774] [Q&A] [answered]
Hello 

I have a doubt about one constraint that I've been fighting for a while. I have to calculate a weights for a portafolio and I have this conditions:
1.  the Total of all weight have to be sum 1.
2.  Each weight cant be bigger than 0.1
3.  The minimum of each weight is  0.001.
4. All the weights between 0.06 and 0.1 cant sum more than 0.5. I mean all the weights that are bigger than 0.05 cant sum more than of 0.4 of 
    the total portafolio.

My problem is the point 4. I've did some tests but I cant find a solution.

This is my example code:

```
import numpy as np
import scipy.sparse as sp
np.random.seed(1)
import cvxpy as cp
mu = np.random.uniform(0,2,size=(10,))
Sigma = np.random.uniform(0,1,size=(10,10))
Sigma = Sigma.T.dot(Sigma)

w = cp.Variable(mu.shape[0])
gamma = cp.Parameter(nonneg=True)
gamma.value=1
ret = mu.T@w 
risk = cp.quad_form(w, Sigma)
prob = cp.Problem(cp.Maximize(ret - gamma*risk), 
               [cp.sum(w) == 1, 
                cp.max(w)<=0.1,
                cp.min(w)>=0.001,
                cp.sum(cp.pos(w-0.05)) <=0.4  #PROBLEM i dont know how to make that condition
                ])
prob.solve()
w.value
```

Thank you in advance for your time.

**Comment by phschiele** [comment, marked as answer]
Hi @LuEdu87 and welcome to the CVXPY!

If you have 10 assets, and each one can have a maximum weight of 10%, then each weight has to be exactly 10% such that the weights will add up to 100%.

I've adapted the numbers a bit such that the constraint you are trying to implement is feasible:
10 assets with a maximum weight of 20% each. All assets > 10% must have a total weight below 50%.

```py
import numpy as np
import cvxpy as cp

np.random.seed(1)
mu = np.random.uniform(0, 2, size=(10,))
Sigma = np.random.uniform(0, 1, size=(10, 10))
Sigma = Sigma.T.dot(Sigma)

w = cp.Variable(mu.shape[0])
is_large_weight = cp.Variable(mu.shape[0], boolean=True)
gamma = cp.Parameter(nonneg=True)
gamma.value = 1
ret = mu.T @ w
risk = cp.quad_form(w, Sigma)


prob = cp.Problem(cp.Maximize(ret - gamma * risk),
                  [cp.sum(w) == 1,
                   w <= 0.2,
                   w >= 0.001,
                   w - 0.1 <= is_large_weight,
                   cp.sum(cp.maximum(w - 0.1, 0)) + cp.sum(is_large_weight) * 0.1 <= 0.5
                   ])
prob.solve()
print(prob.status)
print(w.value)
print(is_large_weight.value)

>>> optimal
>>> [0.1 0.1 0.  0.1 0.2 0.1 0.1 0.1 0.2 0. ]
>>> [0. 0. 0. 0. 1. 0. 0. 1. 1. 0.]
```

**Reply by LuEdu87** [reply]
Thank you!!!  It works perfectly. I'm newbie with cvxpy, and that kind of constraints makes me crazy.

Could you please explain how did you do that? 
1. You created a variable that you dont use in the problem formula. Only in the constraint, and it's  a boolean variable?
2. w-0.1 <= is_large_weight.  it's a kind of "if else"? if the w value is bigger than 0.1 store a 1 in the is_large_weight variable?
3. cp.sum(cp.maximum(w-0.1,0)) --> you sum all the values bigger than 0.1?

if you can recomend a paper/book for newbies like me in cvxpy to understand this kind of examples, I'll appreciate it. (I tried the cvxpy page guide and for me it's really confusing many times.)

Thank you one more time for your solution.
